src/web/mod.rs

```diff
@@ -278,27 +277,23 @@ fn match_version(conn: &Connection, name: &str, version: Option<&str>) -> Option
         rows.map(|row| (row.get(1), row.get(2))).collect()
     };
 
-    // first check for exact match
-    // we can't expect users to use semver in query
-    for version in &versions {
-        if version.0 == req_version {
-            return Some(MatchVersion {
-                corrected_name,
-                version: MatchSemver::Exact(version.clone()),
-            });
-        }
+    // first check for exact match, we can't expect users to use semver in query
+    if let Some((version, id)) = versions.iter().find(|(vers, _)| vers == &req_version) {
```

Comment:
Same as below, I don't see the reason for using `find` or for moving the comment onto one line.

Reply:
We have potentially better optimization from iterators and it's more idiomatic